src/ape_ethereum/ecosystem.py

```diff
@@ -42,6 +46,23 @@
 }
 
 
+class ProxyType(IntEnum):
+    Minimal = 0  # eip-1167 minimal proxy contract
+    Standard = 1  # eip-1967 standard proxy storage slots
+    Beacon = 2  # eip-1967 beacon proxy
+    UUPS = 3  # # eip-1822 universal upgradeable proxy standard
+    Vyper = 4  # vyper <0.2.9 create_forwarder_to
+    Clones = 5  # 0xsplits clones
+    GnosisSafe = 6
+    OpenZeppelin = 7  # openzeppelin upgradeability proxy
+    Delegate = 8  # eip-897 delegate proxy
+
+
+class ProxyInfo(BaseModel):
+    type: ProxyType
+    target: AddressType
```

Comment:
Maybe a way to do this in alignment with `get_proxy_info` in `EcosystemAPI` is to make a `ProxyInfoAPI` object with only `.target` in it, and then here you would subclass and add the type. Really, the target it just what you want when you use it downstream.